core/types/calls.cc

```diff
@@ -712,6 +712,19 @@ DispatchResult dispatchCallSymbol(const GlobalState &gs, DispatchArgs args,
         }
     }
 
+    if (data->isMethodWithKwnil() && !args.args.empty()) {
+        // the called method as explicitely stated no keyword argument with `**nil`
+        auto back = args.args.back()->type.get();
+        if (isa_type<ShapeType>(back) || back->derivesFrom(gs, Symbols::Hash())) {
+            // the method is called with keyword arguments
+            if (auto e = gs.beginError(core::Loc(args.locs.file, args.locs.call),
+                                       errors::Infer::MethodArgumentCountMismatch)) {
+                e.setHeader("No keyword accepted for method `{}`", data->show(gs));
+                e.addErrorLine(method.data(gs)->loc(), "`{}` defined with `{}` here", data->show(gs), "**nil");
+            }
+        }
+    }
```

Comment:
It looks to me like this test case is allowed in Ruby 2.7:

```ruby
def foo(x, **nil)
  p x
end

x = {hello: :world}
foo(x)
```

but I think after the changes in this PR Sorbet will reject this code, right?